File: careerminer-api/src/main/java/com/career_miner/careerminer_api/models/ScrapedData.java
package com.career_miner.careerminer_api.models;

import java.time.Instant;
import java.util.List;

public record ScrapedData(
    String companyName,
    Instant scrapedAt,
    String searchTerm,
    int totalJobs,
    List<FullJobDetails> jobs
) {
    public record FullJobDetails(
        String jobUrl,
        String jobTitle,
        String description,
        String payRange,
        String location,
        String postingDate,
        String jobId
    ) {}
}

File: careerminer-client/app/models/scraped-data.ts
export type FullJobDetails = {
    jobUrl: string,
    jobTitle: string,
    description: string,
    payRange: string,
    location: string,
    postingDate: string,
    jobId: string
}

export type ScrapedData = {
    companyName: string,
    scrapedAt: string,
    searchTerm: string,
    totalJobs: number,
    jobs: FullJobDetails[]
}